docs/developers/authorization.md

```diff
@@ -0,0 +1,107 @@
+# Authorization Set up
+
+  Authorization is the process of giving someone permission to do or have something. In multi-user systems, a system administrator defines for the system which users are allowed access to the system and what privileges of use.
+
+We have a standard access control matrix with 3 axes:
+
+- Actions: CREATE, READ, WRITE, DELETE, PURGE
+- Entities (object): User, Publisher, Package, Package Resource, …
+- Users: a user or type of user
+
+## Introducing Roles
+
+It can be inefficient to list every user for every object. It is also cumbersome to manage in terms of business processes. For example:
+
+Many users in an organization get same set of privileges because of their position in the organization.
+We want to change the permissions associated with a certain level in the organization and to have those permissions changed for all people in that level
+A user may change level frequently (ex. user may get promoted)
+So we create roles
+
+* Per object roles e.g. Package Owner
+* Per system roles e.g. System Administrator
+* A list or algorithm for assigning Users => Roles
+
+Access control algorithm:
+```pytho
+is_allowed(user, entity, action tuple)
+```
+
+For this user: what roles do they have related to this entity and the system?
+Given those roles: what actions do they have: UNIONrole
+Note: it would get more complex if some roles deny access. E.g. Role: Spammer might mean you are denied action to posting etc. Right now we don’t have that issue.
+Is the desired action in that set?
+
+### Roles
+- Package
```

Comment:
I think here you should emphasize these are **examples**. That they should consult XXX source file to see the current list. (This is important as this doc will rapidly go out of date).